### ISSUE TITLE
fromGlobals -> getallheaders dose not exist

### ISSUE THREAD

**Issue by kefahB** [guzzle/psr7#209]
Apache :
Server version: Apache/2.4.33 (Debian)
Architecture:   64-bit

This dose not reproduis on my local server Apache/2.4.33 (Unix) !

In production getallheader function dose not exist : 
```php
var_dump($request->getHeaders()); // got ["Host": "my_server_ip"]
```

then i add this in fromGlobals and i get all my headers !
```php
if (!function_exists('getallheaders'))
        {
            function getallheaders()
            {
                $headers = [];
                foreach ($_SERVER as $name => $value)
                {
                    if (substr($name, 0, 5) == 'HTTP_')
                    {
                        $headers[str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))))] = $value;
                    }
                }
                return $headers;
            }
        }
```

Thanks

**Comment by bkdotcom** [comment]
did you not install guzzle/psr7 via composer??

because if you did, this file would be included automatically:
https://github.com/ralouphie/getallheaders/blob/master/src/getallheaders.php

**Comment by iotch** [comment]
Actually, if `guzzlehttp/guzzle` (6.3.3) is being installed from packagist, the `guzzle/psr7` has the following `composer.json`:

````
{
    "name": "guzzlehttp/psr7",
    "type": "library",
    "description": "PSR-7 message implementation that also provides common utility methods",
    "keywords": ["request", "response", "message", "stream", "http", "uri", "url"],
    "license": "MIT",
    "authors": [
        {
            "name": "Michael Dowling",
            "email": "mtdowling@gmail.com",
            "homepage": "https://github.com/mtdowling"
        },
        {
            "name": "Tobias Schultze",
            "homepage": "https://github.com/Tobion"
        }
    ],
    "require": {
        "php": ">=5.4.0",
        "psr/http-message": "~1.0"
    },
    "require-dev": {
        "phpunit/phpunit": "~4.0"
    },
    "provide": {
        "psr/http-message-implementation": "1.0"
    },
    "autoload": {
        "psr-4": {
            "GuzzleHttp\\Psr7\\": "src/"
        },
        "files": ["src/functions_include.php"]
    },
    "extra": {
        "branch-alias": {
            "dev-master": "1.4-dev"
        }
    }
}
````

So I believe packagist version of guzzle/psr7 was damaged somehow and must be updated.

**Comment by kefahB** [comment]
Hi,

Sorry for the delay of the answer.

Yes in install via composer, but i think there is update is necessary for `guzzlehttp/guzzle` (@iotch )
Actually when I install via composer `composer require guzzlehttp/guzzle` it will install this dependencies in psr7 : 
```
"require": {
        "php": ">=5.4.0",
        "psr/http-message": "~1.0",
},
```
 

But in the version [here](https://github.com/guzzle/psr7/blob/master/composer.json) there is à polyfill `ralouphie/getallheaders`
```
"require": {
        "php": ">=5.4.0",
        "psr/http-message": "~1.0",
        "ralouphie/getallheaders": "^2.0"
},
```
 
On my server based on Debian i have same issu with getallheaders function, and when i put a polyfill I get error in my logs but i dont know if come from guzzle or not !
```
[Tue Aug 14 00:12:30.251102 2018] [proxy_fcgi:error] [pid 2399] [client ---] Premature end of script headers: index.php
[Tue Aug 14 00:12:30.251139 2018] [proxy_fcgi:error] [pid 2399] [client ---] AH01070: Error parsing script headers
[Tue Aug 14 00:12:30.251145 2018] [proxy_fcgi:error] [pid 2399] (22)Invalid argument: [client ---] AH01075: Error dispatching request to :8000
```

**Comment by Tobion** [comment]
The polyfill is not included with the latest release yet. This will be fixed with #200

> AH01070: Error parsing script headers

That is not coming from guzzle. Google finds an apache bug.
